autogpt_platform/frontend/src/components/integrations/credentials-input.tsx

```diff
@@ -95,56 +96,99 @@ export const CredentialsInput: FC<{
   } = credentials;
 
   async function handleOAuthLogin() {
-    const { login_url, state_token } = await api.oAuthLogin(
-      provider,
-      schema.credentials_scopes,
-    );
-    setOAuth2FlowInProgress(true);
-    const popup = window.open(login_url, "_blank", "popup=true");
+    setOAuthError(null);
+    try {
+      const { login_url, state_token } = await api.oAuthLogin(
+        provider,
+        schema.credentials_scopes,
+      );
+      setOAuth2FlowInProgress(true);
+      const popup = window.open(login_url, "_blank", "popup=true");
 
-    const controller = new AbortController();
-    setOAuthPopupController(controller);
-    controller.signal.onabort = () => {
-      setOAuth2FlowInProgress(false);
-      popup?.close();
-    };
-    popup?.addEventListener(
-      "message",
-      async (e: MessageEvent<OAuthPopupResultMessage>) => {
+      if (!popup) {
+        throw new Error(
+          "Failed to open popup window. Please allow popups for this site.",
+        );
+      }
+
+      const controller = new AbortController();
+      setOAuthPopupController(controller);
+      controller.signal.onabort = () => {
+        console.debug("OAuth flow aborted");
+        setOAuth2FlowInProgress(false);
+        popup.close();
+      };
+
+      const handleMessage = async (
+        e: MessageEvent<OAuthPopupResultMessage>,
+      ) => {
+        console.debug("Message received:", e.data);
         if (
           typeof e.data != "object" ||
-          !(
-            "message_type" in e.data &&
-            e.data.message_type == "oauth_popup_result"
-          )
-        )
+          !("message_type" in e.data) ||
+          e.data.message_type !== "oauth_popup_result"
+        ) {
+          console.debug("Ignoring irrelevant message");
           return;
+        }
 
         if (!e.data.success) {
           console.error("OAuth flow failed:", e.data.message);
+          setOAuthError(`OAuth flow failed: ${e.data.message}`);
+          setOAuth2FlowInProgress(false);
+          return;
+        }
+
+        if (e.data.state !== state_token) {
+          console.error("Invalid state token received");
+          setOAuthError("Invalid state token received");
+          setOAuth2FlowInProgress(false);
           return;
         }
 
-        if (e.data.state !== state_token) return;
+        try {
+          console.debug("Processing OAuth callback");
+          const credentials = await oAuthCallback(e.data.code, e.data.state);
+          console.debug("OAuth callback processed successfully");
+          onSelectCredentials({
+            id: credentials.id,
+            type: "oauth2",
+            title: credentials.title,
+            provider,
+          });
+        } catch (error) {
+          console.error("Error in OAuth callback:", error);
+          setOAuthError(
+            `Error in OAuth callback: ${error instanceof Error ? error.message : String(error)}`,
```

Comment:
I believe the "normal" way to do this would be:
```suggestion
            `Error in OAuth callback: ${error instanceof Error ? error.message : error.toString()}`,
```